### PR TITLE
DOC-2774: add cnx string details and RediSearch reqt

### LINKED ISSUE #207: Redis OM .NET documentation
I wanted to suggest to specify in the documentation, that the format of the connection string for Redis.OM is `redis://user:password@host:port`.

It baffled me for quite some time, because I was just trying to paste the connection string from the connection string, given in Azure Cache for Redis settings.

In this case, since the user name is not being used, the format becomes `redis://:password@host:port` (note the colon before password)

By the way, it would also to make sense near the demo app, that RediSearch module is needed for it to work.

### PR DESCRIPTION
**Pull request by dwdougherty**
Resolves #207 .

### CODE REVIEW

File: docs/stack/get-started/om-clients/stack-dotnet.md [hunk 1]
```diff
@@ -41,7 +42,11 @@ Then open the `Redis.OM.Skeleton.csproj` file in your IDE of choice.
 
 ## Configure the app
 
-Add a `"REDIS_CONNECTION_STRING" field to your `appsettings.json` file to configure the application. Set that connection string to be the URI of your Redis instance. If using the docker command mentioned earlier, your connection string will be `redis://localhost:6379`.
+Add a `REDIS_CONNECTION_STRING` field to your `appsettings.json` file to configure the application. Set that connection string to be the URI of your Redis instance. If using the docker command mentioned earlier, your connection string will be `redis://localhost:6379`.
+
+### Connection string specification
```
Comment (slorello89) [inline docs/stack/get-started/om-clients/stack-dotnet.md:47]: Maybe just point at the [Redis URI](https://www.iana.org/assignments/uri-schemes/prov/redis) IANA proposed spec? Or perhaps the one we maintain in the [specs repo](https://github.com/redis/redis-specifications/tree/master/uri)?

**Review by slorello89 — CHANGES_REQUESTED** [review]

File: docs/stack/get-started/om-clients/stack-dotnet.md (hunk 1 above)
Comment (dwdougherty) [reply docs/stack/get-started/om-clients/stack-dotnet.md:47]: Thanks, @slorello89 . The tricky bit is that the colon is required even if there's no username present. It's not clear to me that either of those two references captures that.

[For reference](https://github.com/redis/redis-om-dotnet/blob/968fa767fa7e4362785c4f58905071b23bc0d917/src/Redis.OM/RedisUriParser.cs#L74).

Comment (slorello89) [reply docs/stack/get-started/om-clients/stack-dotnet.md:47]: Technically, those are standard [URI semantics](https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.1) for the Authority section (in this case the user is an empty string). However I see your point, that it is somewhat confusing (probably why I put that error in there :) ). Perhaps link to the URI spec with a note that you need to use a `:password`?

Comment (slorello89) [reply docs/stack/get-started/om-clients/stack-dotnet.md:47]: Also worth noting that `default:password` would also work if there is no username.

Comment (dwdougherty) [reply docs/stack/get-started/om-clients/stack-dotnet.md:47]: Fair enough. I'll do as you ask. Look for another commit soon.

**Review by slorello89 — APPROVED** [review]
LGTM 👍

**Pull request merged**